packages/@o3r/analytics/schematics/index.it.spec.ts

```diff
@@ -2,70 +2,46 @@ import {
   setupLocalRegistry();
-  describe('standalone', () => {
-    beforeAll(async () => {
-      appFolderPath = await prepareTestEnv(appName, 'angular-with-o3r-core');
-      execAppOptions.cwd = appFolderPath;
-    });
-    test('should add analytics to existing application', async () => {
-      packageManagerExec(`ng add --skip-confirmation @o3r/analytics@${o3rVersion}`, execAppOptions);
-
-      packageManagerExec('ng g @o3r/core:component test-component --use-otter-analytics=false', execAppOptions);
-      packageManagerExec('ng g @o3r/analytics:add-analytics --path="src/components/test-component/test-component.component.ts"', execAppOptions);
-      await addImportToAppModule(appFolderPath, 'TestComponentModule', 'src/components/test-component');
-
-      const diff = getGitDiff(appFolderPath);
-      expect(diff.modified).toContain('package.json');
-      expect(diff.added).toContain('src/components/test-component/test-component.analytics.ts');
-
-      expect(() => packageManagerInstall(execAppOptions)).not.toThrow();
-      expect(() => packageManagerRun('build', execAppOptions)).not.toThrow();
-    });
+  beforeAll(async () => {
+    ({ projectPath, workspacePath, projectName, isInWorkspace } = await prepareTestEnv(appFolder));
+    execAppOptions.cwd = workspacePath;
   });
-
-  describe('monorepo', () => {
-    beforeAll(async () => {
-      const workspacePath = await prepareTestEnv(`${appName}-monorepo`, 'angular-monorepo-with-o3r-core');
-      appFolderPath = join(workspacePath, 'projects', 'test-app');
-      execAppOptions.cwd = workspacePath;
-    });
-    test('should add analytics to existing application', () => {
-      // FIXME workaround for pnp
-      packageManagerAdd(`@o3r/analytics@${o3rVersion}`, {...execAppOptions, cwd: appFolderPath});
-
-      const projectName = '--project-name=test-app';
-      packageManagerExec(`ng add --skip-confirmation @o3r/analytics@${o3rVersion} ${projectName}`, execAppOptions);
-
-      packageManagerExec(`ng g @o3r/core:component test-component --use-otter-analytics=false ${projectName}`, execAppOptions);
-      packageManagerExec(
-        'ng g @o3r/analytics:add-analytics --path="projects/test-app/src/components/test-component/test-component.component.ts"',
-        execAppOptions
-      );
-      addImportToAppModule(appFolderPath, 'TestComponentModule', 'src/components/test-component');
-
-      const diff = getGitDiff(execAppOptions.cwd as string);
-      expect(diff.all.some((file) => /projects[\\/]dont-modify-me/.test(file))).toBe(false);
-      expect(diff.modified).toContain('projects/test-app/package.json');
-      expect(diff.added).toContain('projects/test-app/src/components/test-component/test-component.analytics.ts');
-
-      expect(() => packageManagerInstall(execAppOptions)).not.toThrow();
-      expect(() => packageManagerRun('build', execAppOptions)).not.toThrow();
-    });
+  afterAll(async () => {
+    await rm(workspacePath, { recursive: true });
```

Comment:
are we sure this is not executed in case of failed test ?

Reply:
Not sure but it is not important (as not mandatory).
The reason of it is just because I faced an No Space Left in Windows + NPM VM on the last test.
If it is not clean in case of failure, we would have quick IT to fix before reaching the issue

Reply:
I understand the motive, but my concern is that if we clean the folder in case of failure we won't have access to the files in the artifact published at PR level
not blocking though

Reply:
Good point, I will check then

Reply:
It seems that it is not possible in Jest today to avoid executing `afterAll` in case of failure (cf. https://github.com/jestjs/jest/issues/5292)

Reply:
it still possible to work around this like
```typescript
describe('test', () => {
    let wasTestSuccessful = false;

    it('does stuff', () => {
        expect(true).toBe(true);
        wasTestSuccessful = true;
    });

    afterAll(() => {
        if (wasTestSuccessful) {
            try {
                cleanUp();
            } catch (e) {}
        }
    });
});
```
or something like that, the try catch prevents the test to be considered failed if the clean up fails

Reply:
As discussed, this solution is a workaround working only in case of `--bail 1` (which is the value by default) and if we do not have have series of `it`.

If needed it can be put in place but we need to comment its limit (to not have surprise in the future)